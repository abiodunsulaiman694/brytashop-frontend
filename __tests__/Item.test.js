import ItemComponent from "../components/Item";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

const fakeItem = {
  id: "ABX123",
  title: "A cool item",
  price: 5000,
  description: "This item is sooo cool"
};

describe("Snapshot of <Item />", () => {
  const wrapper = shallow(<ItemComponent item={fakeItem} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

describe("<Item />", () => {
  it("renders and displays properly", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const priceTag = wrapper.find("PriceTag");
    expect(priceTag.children().text()).toBe("NGN 50");
    expect(wrapper.find("Title a").text()).toBe(fakeItem.title);
    const img = wrapper.find("img");
    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
  });
  it("renders out the buttons properly", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const buttonList = wrapper.find(".buttonList");
    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find("Link").exists()).toBe(true);
  });
});
