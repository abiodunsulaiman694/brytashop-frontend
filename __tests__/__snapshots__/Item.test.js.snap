// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<ItemStyles__Item>
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABX123",
          },
        }
      }
    >
      <a>
        A cool item
      </a>
    </Link>
  </Title>
  <PriceTag>
    NGNÂ 50
  </PriceTag>
  <p>
    This item is sooo cool
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABX123",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="ABX123"
    />
    <DeleteItem
      id="ABX123"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
